docs/getting-started/how-to-build-a-contract-tx.md

```diff
@@ -0,0 +1,112 @@
+
+The binding between an instance and an input is typically expressed as code like this:
+
+```ts
+instance.to = { tx, inputIndex }
+```
+
+This section could be summarized as the diagram below:
+
+![](../../static/img/contract_tx.svg)
+
+## Tx Builders
+
+We now know that in order to deploy or make calls for contracts, transactions must be built. How can the task be accomplished in code?
+
+
+Thankfully, we have some built-in TX builders for common contracts, so you don't have to worry about how to implement them.
+
+### Deployment Tx Builder
+
+For contract deployment, the built-in Tx builder produces a transaction with the following structure:
+
+* Inputs:
+
+  * [0…]: Several P2PKH inputs for paying transaction fees.
+
+* Outputs:
+
+  * [0]: The output where the contract’s genesis instance comes from.
+  * [1]: A change output if needed.
+
+Because contract deployment transactions can use this structure for any contracts right now, you don’t have to provide your own version.
+
+### Default Call Tx Builder
+
+For contract calls, the built-in default Tx builder produces a transaction with the following structure:
+
+* Inputs
+
+  * [0]: The input that spends the previous contract UTXO with script generated from calling the contract instance’s method.
+  * [1…]: Several P2PKH inputs for paying transaction fees.
+
+* Outputs
+
+  * [0…N-1]: Several outputs that each contain a new contract instance (UTXO) if the contract is stateful.
+  * [N]: A change output if needed.
+
+
+### Customized Call Tx builder
+
+If the default Tx builder does not meet your specific requirements, such as having other specified inputs or outputs in your Tx, you can define a customized Tx builder for a public method of your contract.
+
+Usually you can define a customized Tx builder for a contract @method like this:
+
+```ts
+// bind a customized tx builder for `MyContract.unlock`
+MyContract.bindTxBuilder("unlock", (options, ...args) => { 
+
+  let result: Promise<BuildMethodCallTxResult<MyContract>>;
+
+  const current = options.current;
+
+  const unsignedTx: bsv.Transaction = new bsv.Transaction()
+    // add contract input
+    .addInput(current.buildContractInput(options.fromUTXO))
+    // add p2pkh inputs
+    .from(options.utxos)
+    // add an extra p2pkh output
+    .addOutput(new bsv.Transaction.Output({
+        script: bsv.Script.fromHex(Utils.buildPublicKeyHashScript(args[0])),
+        satoshis: args[1]
+    }))
+    // add change output
+    .change(options.changeAddress);
+
+  result = {
+    unsignedTx,
+    atInputIndex: 0
+  };
+
+  return Promise.resolve(result)         
+})
+```
+
+Please be aware that your customized Tx builder must always deliver a transaction that adheres to your public method's logic and the hash type of its signature. More information can be found in [this section](./what-is-scriptcontext.md).
```

Comment:
Just
`Please be aware that your customized Tx builder must always produce a transaction that adheres to your public method's logic, i.e., making all assertions executed in it pass.`

No need to mention a specific case of hash type or ScriptContext.